io/net/ethtool_test.py

```diff
@@ -102,19 +110,41 @@ def test_ethtool(self):
                               ignore_status=True)
             if ret.exit_status != 0:
                 self.fail("failed")
-        if not self.ping_check('-c 5'):
-            self.fail("ping failed after interface is up")
+        if not HostInfo.ping_check(self, self.iface, self.peer, '10000',
+                                   flood=True):
+            self.fail("flood ping test failed")
 
-    def ping_check(self, options):
+    def test_ethtool_toggle_priv_flags(self):
         '''
-        Checks if ping to peer works fine and returns True.
-        Returns False otherwise.
+        Toggle the priv flag settings of the driver.
         '''
-        cmd = "ping -I %s %s %s" % (self.iface, options, self.peer)
-        if process.system(cmd, shell=True, verbose=True,
-                          ignore_status=True) != 0:
-            return False
-        return True
+        priv_pass = []
+        priv_fail = []
+        cmd = "ethtool --show-priv-flags %s" % (self.iface)
+        ret = process.run(cmd, shell=True, verbose=True,
+                          ignore_status=True)
+        if ret.exit_status:
+            self.cancel("Device Doesn't support Private flags")
+        for line in ret.stdout.decode("utf-8").splitlines():
+            if "off" in line:
+                val = "on"
+            else:
+                val = "off"
+            if "flags" not in line:
+                priv_flag = line.split(':')[0]
+                cmd = "ethtool --set-priv-flags %s \"%s\" %s" % \
+                      (self.iface, priv_flag.rstrip(), val)
+                ret1 = process.run(cmd, shell=True, verbose=True,
+                                   ignore_status=True)
+                if ret1.exit_status:
+                    priv_fail.append(priv_flag.rstrip())
+                else:
+                    priv_pass.append(priv_flag.rstrip())
```

Comment:
This is not needed

Reply:
@narasimhan-v  eventhough we dont explicitly print this.   in job.log actually gives us the pass and fail output like below.   Usefull to get the list of what priv_flags could be set/unset basically passed and failed.

2019-11-27 20:04:18,968 test             L0869 DEBUG|  -> cmd <class 'str'>: ethtool --set-priv-flags eth3 "tx_xdp_hw_checksum" off
2019-11-27 20:04:18,968 test             L0869 DEBUG|  -> priv_fail <class 'list'>: ['xdp_tx_mpwqe', 'dropless_rq']
2019-11-27 20:04:18,968 test             L0869 DEBUG|  -> priv_pass <class 'list'>: ['rx_cqe_moder', 'tx_cqe_moder', 'rx_cqe_compress', 'rx_striding_rq', 'rx_no_csum_complete', 'sniffer', 'per_channel_stats', 'tx_xdp_hw_checksum']

Reply:
@manvanthar Ah ok, agreed.